lib/doorkeeper/config.rb

```diff
@@ -292,24 +292,32 @@ def option(name, options = {})
     attr_reader :api_only
     attr_reader :enforce_content_type
 
+    def api_only
+      @api_only ||= false
+    end
+
+    def enforce_content_type
+      @enforce_content_type ||= false
+    end
+
     def refresh_token_enabled?
-      defined?(@refresh_token_enabled) && @refresh_token_enabled
+      !!(defined?(@refresh_token_enabled) && @refresh_token_enabled)
```

Comment:
I think in this case we can return to previous  `!!@refresh_token_enabled` solution, but what is the case? `nil`-values in ruby is always falsey, what is the purpose of this changes?

Reply:
The point is to avoid tri-state booleans as described here: https://robots.thoughtbot.com/avoid-the-threestate-boolean-problem. (The article is about DB, but I apply the same rule for Ruby code. 
Also as a rule of thumb, all question-marks methods should return either `true` or `false`. Very opinionated though, I can revert it if you're not comfortable :).

Reply:
No no, I think it's a reasonable one. So maybe we can just return `!!@var` syntax for the options?

Reply:
I'm not sure I get what you are suggesting. 

Can you provide an example?

Reply:
Previously it was just

```ruby
!!@refresh_token_enabled
```

instead of

```ruby
defined?(@refresh_token_enabled) && @refresh_token_enabled
```

Reply:
A little bit another way :)

https://github.com/doorkeeper-gem/doorkeeper/blob/ed95d941e5846ed5b8ef90ff2f6bac96f83e2fec/lib/doorkeeper/config.rb#L284-L285

So maybe we can save the code from your commit

Reply:
Ok I get what you meant.

Reply:
Ok I get what you meant. The issue with this approach is that the Doorkeeper.configuration object returned will contains the wrong value, it can be misleading to have the getter methods returning false and the Configuration object having `nil`.

Reply:
I think we can save you changes

Reply:
So, do I have something to change or not? 
Sorry, I'm super confused on what you need me to do, and it's also the end of the day in France ;)